pkg/executor/sortexec/sort.go

```diff
@@ -336,11 +518,96 @@ func (e *SortExec) fetchRowChunks(ctx context.Context) error {
 		return err
 	}
 
-	e.sortPartitions = append(e.sortPartitions, e.curPartition)
+	e.Unparallel.sortPartitions = append(e.Unparallel.sortPartitions, e.curPartition)
 	e.curPartition = nil
 	return nil
 }
 
+func (e *SortExec) fetchChunksParallel(ctx context.Context) error {
+	// Wait for the finish of all workers
+	workersWaiter := util.WaitGroupWrapper{}
+	// Wait for the finish of chunk fetcher
+	fetcherWaiter := util.WaitGroupWrapper{}
```

Comment:
Seems they can be combined into one.

Reply:
> Seems they can be combined into one.

We can't combine them into one, as when one worker panics all workers may exit without consuming chunks sent by fetcher, this will lead to hang in sort spill.